indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -274,11 +286,6 @@ public TaskLock lock(final Task task, final Interval interval) throws Interrupte
             log.makeAlert("Same Task is trying to acquire lock for overlapping interval")
                .addData("task", task.getId())
                .addData("interval", interval);
-            throw new ISE(
```

Comment:
@jihoonson Overlapping intervals causes deadlock for `lockAquireAction` and not for `lockTryAquireAction`. I think throwing exception for `tryLock`  is unnecessary. Other problem I could see is in the SegmentAllocationAction, https://github.com/druid-io/druid/blob/master/indexing-service/src/main/java/io/druid/indexing/common/actions/SegmentAllocateAction.java?utf8=%E2%9C%93#L177. SegmentAllocateAction tries to acquire lock for the same task with different intervals and throwing exception for overlapping interval in `tryLock` might cause failure of segment allocations. 
For now I'm just alerting and logging which might be useful if someone run into deadlock issue.